documentation/source/benchmark.rst

```diff
@@ -0,0 +1,325 @@
+Benchmarking Coreset Algorithms
+===============================
+
+In this benchmark, we assess the performance of four different coreset algorithms:
+:class:`~coreax.solvers.KernelHerding`, :class:`~coreax.solvers.SteinThinning`,
+:class:`~coreax.solvers.RandomSample`, and :class:`~coreax.solvers.RPCholesky`.
+Each of these algorithms is evaluated across four different tests, providing a
+comparison of their performance and applicability to various datasets.
+
+Test 1: Benchmarking Coreset Algorithms on the MNIST Dataset
+------------------------------------------------------------
+
+The first test evaluates the performance of the coreset algorithms on the
+**MNIST dataset** using a simple neural network classifier. The process follows
+these steps:
+
+1. **Dataset**: The MNIST dataset consists of 60,000 training images and 10,000
+   test images. Each image is a 28x28 pixel grey-scale image of a handwritten digit.
+
+2. **Model**: A Multi-Layer Perceptron (MLP) neural network is used for
+   classification. The model consists of a single hidden layer with 64 nodes.
+   Images are flattened into vectors for input.
+
+3. **Dimensionality Reduction**: To speed up computation and reduce dimensionality, a
+   density preserving :class:`~umap.umap_.UMAP` is applied to project the 28x28 images into 16 components
+   before applying any coreset algorithm.
+
+4. **Coreset Generation**: Coresets of various sizes are generated using the
+   different coreset algorithms. For :class:`~coreax.solvers.KernelHerding` and
+   :class:`~coreax.solvers.SteinThinning`, :class:`~coreax.solvers.MapReduce` is
+   employed to handle large-scale data.
+
+5. **Training**: The model is trained using the selected coresets, and accuracy is
+   measured on the test set of 10,000 images.
+
+6. **Evaluation**: Due to randomness in the coreset algorithms and training process,
+   the experiment is repeated 5 times with different random seeds. The benchmark is run
+   on an **Amazon g4dn.12xlarge instance** with 4 NVIDIA T4 Tensor Core GPUs, 48 vCPUs,
+   and 192 GiB memory.
+
+**Results**:
+The accuracy of the MLP classifier when trained using the full MNIST dataset
+(60,000 training images) was 97.31%, serving as a baseline for evaluating the performance
+of the coreset algorithms.
+
+- Plots showing the accuracy (with error bars) of the MLP's predictions on the test set,
```

Comment:
This bullet point is silly. Remove it and dedent all the remaining lines in this section.